super-mode-calculator/src/lib/build-query.ts

```diff
@@ -0,0 +1,154 @@
+import { format, subDays, subHours } from 'date-fns';
+import { regionSql } from '../lambdas/query/regionSql';
+import {
+	SUPER_MODE_MINIMUM_AV,
+	SUPER_MODE_MINIMUM_VIEWS,
+	SUPER_MODE_WINDOW_IN_HOURS,
+} from './constants';
+import { toDateHourString, toDateString } from './date';
+
+export interface BigQueryResult {
+	url: string;
+	region: string;
+	total_av: number;
+	total_views: number;
+	av_per_view: number;
+}
+
+export const buildQueryForSuperMode = (
+	stage: 'CODE' | 'PROD',
+	now: Date = new Date(),
+) => {
+	const start = subHours(now, 1);
+	const windowStartDate = subHours(start, SUPER_MODE_WINDOW_IN_HOURS);
+
+	const dateString = toDateString(windowStartDate);
+	const dateHourString = toDateHourString(windowStartDate);
+	const dateForCurrencyConversionTable = subDays(windowStartDate, 1); //This table is updated daily  but has a lag of 1 day
+
+	return `
+WITH
+acquisitions_with_regions AS (SELECT *,${regionSql('country_Code')}
+		FROM datatech-platform-${stage.toLowerCase()}.datalake.fact_acquisition_event
+		WHERE
+			DATE (event_timestamp) >= '${dateString}'
+		  AND
+			event_timestamp >= TIMESTAMP '${dateHourString}'),
```

Comment:
Just noticed we don't have end timestamps for the acquisitions + views queries.
This was also the case in the old super mode query, though I guess it wasn't much of an issue because it was querying up to the nearest hour.
The end timestamp should be `toDateHourString(start)`
And actually, `start` should be renamed to `end`!

Reply:
Oh , yes, this was something I was not sure about.